src/components/JTextArea/JTextArea.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import React, { PureComponent } from 'react'
+import { omit } from 'lodash-es'
+import classNames from 'classnames'
+
+import jTextAreaStyle from './jTextArea.m.scss'
+
+type Theme = 'blue'
+
+type Props =
+  StyleComponent<Theme>
+  & {
+    onChange: Function,
+    onBlur?: Function,
+    onFocus?: Function,
+  }
+
+const MAX_ROWS = 12
+
+function heightCalc({ currentTarget: target }: SyntheticEvent<HTMLTextAreaElement>): void {
+  if (target) {
+    while (target.clientHeight < target.scrollHeight && target.rows <= MAX_ROWS) {
+      // mutating the DOM
+      // eslint-disable-next-line fp/no-mutation, no-param-reassign
+      target.rows += 1
+    }
+  }
+}
+
+export class JTextArea extends PureComponent<Props, void> {
```

Comment:
Maybe we should also rename it to `JAutoExpandTextArea` or something like this, so it would not interfere with basic TextArea without auto expand function.